src/data_formatter.ts

```diff
@@ -240,3 +240,108 @@ export default class DataFormatter {
     }
   }
 }
+
+export function containsWideDataFrame(data: any[]) {
+  if (!data || !data.length) {
+    return false;
+  }
+
+  return data.some((d) => d && d.hasOwnProperty('type') && d.type === 'table');
+}
+
+interface LegacyTable {
+  columns: { text: string }[];
+  type: string;
+  refId: string;
+  meta: any;
+  rows: any[][];
+}
+
+interface LegacyTimeSeries {
+  alias: string;
+  target: string;
+  datapoints: any[];
+  refId: string;
+  meta: any;
+}
+
+interface FrameMapInfo {
+  timeColumn: number;
+  columnMap: Map<string, LegacyTimeSeries>;
+  refId: string;
+  meta: any;
+}
+
+export function fromWideDataFrame(data: LegacyTable[]): LegacyTimeSeries[] {
```

Comment:
thought this was going to be part of toLegacyResponseData https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/dataframe/processDataFrame.ts#L320 ? Having it here means this problem is still there for other panels.  But I guess it's because your waiting on https://github.com/grafana/grafana-plugin-sdk-go/pull/375

Reply:
yes, you're correct!